src/drivers/imx/edma.c

```diff
@@ -333,8 +333,8 @@ static int edma_set_config(struct dma_chan_data *channel,
 			   struct dma_sg_config *config)
 {
 	int handshake, irq;
-	uint16_t soff = 0;
-	uint16_t doff = 0;
+	int16_t soff = 0;
+	int16_t doff = 0;
 
 	/* We may need to pass some data through the handshake in the
```

Comment:
offset should really be unsigned. Can't we just fix the call to SGN?

Reply:
We may consider #2067 applies to SGN (though AFAIK the hardware can support negative offsets, we just don't use them)